### PR TITLE
Improvements to edit model server modal

### LINKED ISSUE #1626: Edit Model server modal enhancements
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Deploy type

OpenDataHub core version (eg. `v1.6.0`)

### Version

v2.13.0

### Current Behavior

When you click `Edit model server`, the `Add model server` button shows `Add` enabled with the field values intact. This is what I see after clicking on `Edit model server`
Screenshot 2023-08-01 at 7 23 41 PM

### Expected Behavior

The `Add model server` should be changed to `Edit model server` and the button should be changed to `Update`, The `Update` button should only be enabled when one or more fields are updated in the modal(consistent with the other modals)

### Steps To Reproduce

1. Go to `Models and Model servers` in DS projects
2. Click on the `Edit model server` button from the Kebab menu(You may have to add a model server and deploy model if you haven't done that already, so that you can "edit" it)

### Workaround (if any)

_No response_

### What browsers are you seeing the problem on?

_No response_

### Anything else

_No response_

### PR DESCRIPTION
**Pull request by DaoDaoNoCode**
Closes #1626 

## Description

1. Update the title and button label when editing the model server

Screenshot 2023-09-28 at 10 52 22 AM

2. Disable the update button when there is no change in the fields
3. Automatically disable the tokens when the last token is removed

https://github.com/opendatahub-io/odh-dashboard/assets/37624318/a7b28de9-a914-4779-9acc-16110d1ee57f

## How Has This Been Tested?

1. Edit an existing model server
2. Check the title and the button label
3. Try to change all the fields and change them back, see the enabled/disabled status of the update button

## Test Impact

Add an integration test for editing the model server to check the enabled/disabled status when changing the inputs fields

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Comment by DaoDaoNoCode** [comment]
@vconzola Can you check the UI change and the behavior in the description?

**Review by Gkrumbach07 — APPROVED** [review]
/lgtm tests look good and good seperation of utils

**Review by manaswinidas — APPROVED** [review]
/lgtm

Tested. Works fine.

**Comment by andrewballantyne** [comment]
/hold

This is related to a story -- please rebase onto the proper feature branch

**Comment by DaoDaoNoCode** [comment]
/unhold
Changed the target to the feature branch and rebased.

File: frontend/src/pages/modelServing/utils.ts [hunk 1]
```diff
@@ -114,3 +131,51 @@ export const getTokenNames = (servingRuntimeName: string, namespace: string) =>
...
+
+export const getServingRuntimeTokens = (tokens?: SecretKind[]): ServingRuntimeToken[] =>
+  (tokens || []).map((secret) => ({
+    name: getDisplayNameFromK8sResource(secret) || secret.metadata.name,
+    editName: secret.metadata.name,
+    uuid: secret.metadata.name,
+    error: '',
+  }));
+
+export const isModelServerEditInfoChanged = (
```
Comment (Gkrumbach07) [inline frontend/src/pages/modelServing/utils.ts:161]: this needs to check for accelerator dropdown now too.

**Review by Gkrumbach07 — CHANGES_REQUESTED** [review]

File: frontend/src/pages/modelServing/utils.ts (hunk 1 above)
Comment (DaoDaoNoCode) [reply frontend/src/pages/modelServing/utils.ts:196]: Updated, could you check it again?

File: frontend/src/pages/modelServing/utils.ts [hunk 2]
```diff
@@ -114,3 +133,86 @@ export const getTokenNames = (servingRuntimeName: string, namespace: string) =>
...
+    name: getDisplayNameFromK8sResource(secret) || secret.metadata.name,
+    editName: secret.metadata.name,
+    uuid: secret.metadata.name,
+    error: '',
+  }));
+
+export const getAcceleratorGpuCount = (
+  accelerator: AcceleratorKind,
+  resources: ContainerResources,
+) => Number(resources.requests?.[accelerator.spec.identifier] ?? 0);
```
Comment (Gkrumbach07) [inline frontend/src/pages/modelServing/utils.ts:166]: Accelerators are not model serging specific. can you move this util function?

Comment (DaoDaoNoCode) [reply frontend/src/pages/modelServing/utils.ts:166]: Can you tell me what's the best location for this function?

Comment (Gkrumbach07) [reply frontend/src/pages/modelServing/utils.ts:166]: `frontend/src/utilities/utils.ts`

**Review by Gkrumbach07 — APPROVED** [review]
/lgtm

**Review by lucferbux — APPROVED** [review]
/lgtm

**Pull request merged**